schema/double.go

```diff
@@ -13,19 +43,22 @@ func writeDouble(r float64, w io.Writer) error {
 }
 `
 
+// DoubleField represents a float(64) Avro field.
 type DoubleField struct {
 	PrimitiveField
 }
 
+// NewDoubleField constructs a new `DoubleField` for the given definition
 func NewDoubleField(definition interface{}) *DoubleField {
 	return &DoubleField{PrimitiveField{
 		definition:       definition,
 		name:             "Double",
-		goType:           "float64",
+		goType:           reflect.Float64.String(),
```

Comment:
I like this, at least for primitive types this is a clear win over hard-coded strings IMO.